### DISCUSSION TITLE
Timeout des Verbindungsversuchs zur Vitoconnect 100

### DISCUSSION THREAD

**Discussion by CM000n** [philippoo66/optolink-splitter#27] [Q&A] [answered]
Hallo zusammen. Vorab noch schon einmal vielen Dank für dieses großartige Projekt!

Ich habe heute versucht, den Optolink Splitter mit einem Minimal Setup in Verbindung mit meiner Vitocrossal 300 und einer Vitoconnect 100 einzurichten.
Ich habe dafür den aktuellen `main` Branch genutzt und als `poll_items` lediglich die Aussen- und Kesseltemperatur angegeben. Das scheint soweit auch zu funktionieren und die beiden Temperaturwerte werden korrekt erkannt und per mqtt gepuplished.

Das ganze funktioniert so ohne Angabe einer `port_vitoconnect` Adresse. Wenn ich als `port_vitoconnect` Adresse nun jedoch `'/dev/ttyAMA0'` angebe und, entsprechend der Anleitung, warte bis `awaiting VS2...` angezeigt und dann die Vitoconnect anstecke, passiert weiter nichts. Die Vitoconnect startet und dann blinkt nach einiger Zeit die rote LED, um einen Fehler zu signalisieren. Das `optolinkvs2_switch.py` Skript läuft dabei dann in den in der Config angegebenen `vs2timeout`.

Bei meiner Hardware handelt es sich um einen RaspberryPi 4 B, bei dem ich die boot firmware config entsprechend dieser Anleitung hier (https://github.com/philippoo66/optolink-splitter/wiki/520-termios.error:-(22,-'Invalid-argument')) bearbeitet habe. 
Bei meinem UART USB Adapter handelt es sich um diesen hier (https://www.berrybase.de/konverter-usb-typ-a-stecker-uart-cp2102), den ich (hoffentlich) richtig mit GND, RX & TX am Raspi verbunden habe. Der Jumper stand bereits richtig auf 3.3V.

Leider fehlt mit ein wenig Erfahrung mit der Materie, um das ganze vernünftig debuggen zu können. Deswegen dachte ich einfach, ich schreibe hier mal in die Diskussions.

Vielen Dank schon mal Vorab!

..::Nachtrag::..
Ich habe das "Gefühl", als würde die Vitoconnect irgendwie den RassPi abschießen.
Zumindest war dieser jetzt schon zwei mal, kurz nach dem ich die Vitoconnect angeschaltet habe, plötzlich nicht mehr erreichbar. Eventuell kann ich das Verhalten morgen nochmal validieren.
..::Nachtrag zum Nachtrag::..
Das "abschießen" des Raspberrys scheint nicht durch das anschließen der Vitoconnect verursacht worden zu sein. Vielmehr verliert der Raspberry einfach sporadisch seine WLAN Verbindung. Ich habe nun für die WLAN Karte das Power Management deaktiviert. Dadurch scheint sich das Verhalten verbessert zu haben.

**Comment by philippoo66** [comment, marked as answer]
Moin!

da fallen mir zuerst 2 Sachen zu ein:

1. hast du bei Anschluss des CP2102 Module RX und TX gekreuzt? das was der eine sagt, muss der andre empfangen und umgekehrt
2. es gibt ein kleines Skript list_ports.py im Projekt. Führ das mal aus und schau, ob da der AMA0 auftaucht (wobei wenn das nicht korrekt wäre, gäbe es wohl eine Fehlermeldung al la kann Port nicht öffnen)

Wenn beides nicht hilft, musst du mal in der settings_ini das `log_vitoconnect` auf True setzen und wir müssen und das anschauen.

**Reply by philippoo66** [reply]
ach nochwas, weil es letztens grad da war:

du musst in raspi-config die serielle Console disablen und den seriellen port (für Apps) enablen - hast du das gemacht? (muss ich noch in's Wiki/Readme einpflegen...)

https://community.viessmann.de/t5/Waermepumpe-Hybridsysteme/Datenpunkte-in-der-API-gesucht/m-p/459215#M122613

**Reply by CM000n** [reply]
> ach nochwas, weil es letztens grad da war:
> 
> du musst in raspi-config die serielle Console disablen und den seriellen port (für Apps) enablen - hast du das gemacht? (muss ich noch in's Wiki/Readme einpflegen...)
> 
> https://community.viessmann.de/t5/Waermepumpe-Hybridsysteme/Datenpunkte-in-der-API-gesucht/m-p/459215#M122613

Vielen Dank. Dass war tatsächlich nicht entsprechend eingestellt. Hat aber keine Änderung gebracht.

Zu 1. Anschlüsse habe ich nochmal kontrolliert. Sollten passen
Zu 2. `list_ports.py` gibt folgenden Output:
```
Verfügbare serielle Ports:
/dev/ttyAMA0: ttyAMA0 [fe201000.serial]
/dev/ttyUSB0: CP2102 USB to UART Bridge Controller - CP2102 USB to UART Bridge Controller [USB VID:PID=10C4:EA60 SER=0001 LOCATION=1-1.4]
```

Das logging der Vitoconnect habe ich bei `log_vitoconnect` mal auf `True` gesetzt. Es wird dadurch auch eine `vitolog.txt` Datei erstellt. Diese ist jedoch leer.

**Reply by CM000n** [reply]
Oh Mann, ich Dussel
Habe es falsch gelesen. Natürlich war TX auf TX und RX auf RX gesteckt 🙄
Nachdem ich diese überkreuzt habe, funkt es munter hin uns her und die Vitoconnect schaltet auf grün.

Sry!

Aber vll. sollte man das mit dem Überkreuzen der Anschlüsse für Döspaddel wie mich noch mit in die Beschreibung/das Wiki übernehmen?
Falls es schon drin stehen sollte und ich es überlesen habe, natürlich mea culpa.

**Reply by philippoo66** [reply]
danke für die Rückmeldung! freut mich, dass es jetzt geht!

'habe es im ReadMe/Wiki ergänzt.

**Reply by philippoo66** [reply]
wenn es jetzt läuft, schalte das log_vitoconnect besser wieder aus (es sei denn, du möchtest wissen, welche Datenpunkte Viessmann so liest...) - das vitolog.txt wird recht schnell ziemlich gross....
